File: app/src/main/java/tech/leonam/resultadodocertame/view/Logar.java
package tech.leonam.resultadodocertame.view;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

import tech.leonam.resultadodocertame.R;

public class Logar extends AppCompatActivity {
    private Button logar, recuperarSenha;
    private EditText email, senha;
    private TextView cadastro;
    private FirebaseAuth mAuth;
    private FirebaseUser user;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_logar);
        iniciarComponentes();
        logar();
        cadastrar();
        recuperarSenha();
    }

    public void iniciarComponentes() {
        logar = findViewById(R.id.botaoLogar);
        recuperarSenha = findViewById(R.id.botaoRecuperarSenha);
        email = findViewById(R.id.emailLogar);
        senha = findViewById(R.id.senhaLogar);
        cadastro = findViewById(R.id.cadastrar);
        mAuth = FirebaseAuth.getInstance();
    }
    @Override
    protected void onStart() {
        super.onStart();
        var currentUser = mAuth.getCurrentUser();
        if (currentUser != null) {
            abrirMenuPrincipal();
        }
    }

    public void logar() {
        logar.setOnClickListener(e -> mAuth.signInWithEmailAndPassword(email.getText().toString(), senha.getText().toString()).addOnCompleteListener(this, task -> {
            boolean tarefa = task.isSuccessful() ? abrirMenuPrincipal() : abrirErroDeLogin();
        }));
    }

    public boolean abrirMenuPrincipal() {
        startActivity(new Intent(this, MainActivity.class));
        return true;
    }

    public boolean abrirErroDeLogin() {
        var dialog = new AlertDialog.Builder(Logar.this);
        dialog.setTitle("Infelizmente não foi possível efetuar login");
        dialog.setMessage("Verifique seu Email ou Senha.");
        dialog.create().show();
        return true;
    }

    public void cadastrar() {
        cadastro.setOnClickListener(e-> startActivity(new Intent(this,Cadastro.class)));
    }

    public void recuperarSenha() {
        recuperarSenha.setOnClickListener(e -> {
            //TODO janela nova pra abrir
        });
    }
}

File: app/src/main/java/tech/leonam/resultadodocertame/view/activitys/Logar.kt
package tech.leonam.resultadodocertame.view.activitys

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.google.android.gms.tasks.Task
import com.google.firebase.auth.AuthResult
import com.google.firebase.auth.FirebaseAuth
import tech.leonam.resultadodocertame.databinding.ActivityLogarBinding

class Logar : AppCompatActivity() {
    private var mAuth: FirebaseAuth? = FirebaseAuth.getInstance()
    private lateinit var binding: ActivityLogarBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityLogarBinding.inflate(layoutInflater)
        setContentView(binding.root)
        supportActionBar!!.hide()
        window.statusBarColor = Color.BLACK
        logar()
        cadastrar()
        recuperarSenha()
    }

    override fun onStart() {
        super.onStart()
        if (mAuth?.currentUser != null) abrirMenuPrincipal()
    }

    private fun logar() {
        binding.botaoLogar.setOnClickListener {
            mAuth!!.signInWithEmailAndPassword(
                binding.emailLogar.text.toString(), binding.senhaLogar.text.toString()
            ).addOnCompleteListener(this) { task: Task<AuthResult?> ->
                if (task.isSuccessful) abrirMenuPrincipal() else abrirErroDeLogin()
            }
        }
    }

    private fun abrirMenuPrincipal(): Boolean {
        startActivity(Intent(this, MainActivity::class.java))
        return true
    }

    private fun abrirErroDeLogin(): Boolean {
        val dialog = AlertDialog.Builder(this@Logar)
        dialog.setTitle("Infelizmente não foi possível efetuar login")
        dialog.setMessage("Verifique seu Email ou Senha.")
        dialog.create().show()
        return true
    }

    private fun cadastrar() {
        binding.cadastrar.setOnClickListener {
            startActivity(
                Intent(
                    this,
                    Cadastro::class.java
                )
            )
        }
    }

    private fun recuperarSenha() {
        binding.botaoRecuperarSenha.setOnClickListener{}
    }
}